SPECS/rust/rust.spec

```diff
@@ -114,6 +118,7 @@ export CXXFLAGS="`echo " %{build_cxxflags} " | sed 's/ -g//'`"
 sh ./configure \
```

Comment:
in the BuildRequires: above can we take an explicit dependency on the libz and in the vendor tarball generation for this can we explicitly remove any vendored source for zlib.  We want to make sure that rust takes a dependency on the system zlib and not the internally vendored one.  I believe this is already working implicitly, goal is to make it explicit
[Bug 44559508](https://microsoft.visualstudio.com/OS/_workitems/edit/44559508): [2.0][3.0] [Rust] Remove vendored system libs from uploaded source tarball